core/src/main/java/org/hisp/dhis/android/core/option/OptionSetModel.java

```diff
@@ -38,23 +39,44 @@
 import com.google.auto.value.AutoValue;
 
 import org.hisp.dhis.android.core.common.BaseIdentifiableObjectModel;
+import org.hisp.dhis.android.core.common.StatementBinder;
 import org.hisp.dhis.android.core.common.ValueType;
 import org.hisp.dhis.android.core.data.database.DbValueTypeColumnAdapter;
+import org.hisp.dhis.android.core.utils.Utils;
 
+import static org.hisp.dhis.android.core.utils.StoreUtils.sqLiteBind;
 
 @AutoValue
-public abstract class OptionSetModel extends BaseIdentifiableObjectModel {
+public abstract class OptionSetModel extends BaseIdentifiableObjectModel implements StatementBinder {
```

Comment:
This class is not generic right? Also, shouldn't it belong to the Tracker domain?

Reply:
The class is not generic, but it contains the additional properties that we have added to the model classes in the aggregate domain to be able to use them with our generic classes. To be able to write integration tests for our generic classes, we had to pick one model class and add these additional properties (in the best case a simple one), because the table should exist in the system during the execution of the integration test. 

It doesn't have any implications in performance or whatsoever, but if you don't want us to touch any class from the Tracker domain, we can leave it out of this pull request and include the tests using DataSet in the next pull request.